crypto/tls/test/tls_integration_test.go

```diff
@@ -145,14 +145,14 @@ func newIntegrationClientServer(
 				if err == nil {
 					return false
 				}
-				return strings.Contains(err.Error(), "connection reset by peer") || strings.Contains(err.Error(), "broken pipe")
+				return strings.Contains(err.Error(), "connection reset by peer") ||
+					strings.Contains(err.Error(), "broken pipe") ||
+					strings.Contains(err.Error(), "client conn could not be established")
 			}
-			for i := 0; i < 3 && isRST(err) && tc.httpExpectError != nil; i++ {
+			for i := 0; i < 5 && isRST(err) && tc.httpExpectError != nil; i++ {
+				t.Logf("Sleeping before retry #%d, due to RST error: %s", i+1, err)
 				time.Sleep(100 * time.Millisecond)
 				resp, err = client.Do(req)
-				if err == nil {
-					defer resp.Body.Close()
-				}
```

Comment:
Why this change? `resp` get overwritten each loop cycle. I think it was correct being here. What I think that can be removed is the `defer resp.Body.Close()` below, not this one.

Reply:
I actually thought this through, and I'm pretty certain I've got it right. If `err == nil`, the loop breaks because the `isRST(err)` condition returns false. Ergo, in that case execution continues on to line 157 and executes `defer resp.Body.Close()`.

Reply:
If I should turn out to be wrong in my analysis (I don't see how, as I've looked it over again), we can fix in a follow-up PR :)